kustomize/commands/edit/fix/convert.go

```diff
@@ -134,7 +135,7 @@ func getNodesFromFile(fileName string, fSys filesys.FileSystem) ([]*kyaml.RNode,
 	}
 	out := &bytes.Buffer{}
 	r := kio.ByteReadWriter{
-		Reader:                bytes.NewBufferString(string(b)),
+		Reader:                bytes.NewReader(b),
```

Comment:
```suggestion
		Reader:                bytes.NewBuffer(b),
```

In https://github.com/kubernetes-sigs/kustomize/pull/5211/files#r1247400428 you explain that we don't want to convert to string and back to byte, which is fine, but it is not clear why we need to switch from `Buffer` to `Reader`. 

If there is some reason to switch to `Reader`, I would prefer that happen in its own PR with a description, to avoid conflating it with the nil pointer error change.

Reply:
Using `NewBuffer` instead